control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2975,12 +3016,29 @@ func (r *HostedControlPlaneReconciler) reconcileKubeAPIServer(ctx context.Contex
 		return fmt.Errorf("failed to reconcile localhost kubeconfig secret: %w", err)
 	}
 
-	externalKubeconfigSecret := manifests.KASExternalKubeconfigSecret(hcp.Namespace, hcp.Spec.KubeConfig)
-	if _, err := createOrUpdate(ctx, r, externalKubeconfigSecret, func() error {
+	// Generate the new externalKubeconfigSecret if the external name is set
+	if len(hcp.Spec.KasCustomKubeconfig.KasDNSName) > 0 {
+		if hcp.Spec.Configuration != nil && hcp.Spec.Configuration.APIServer != nil && len(hcp.Spec.Configuration.APIServer.ClientCA.Name) > 0 {
+			rootCA = manifests.KASExternalCAConfigMap(hcp.Spec.Configuration.APIServer.ClientCA.Name)
```

Comment:
ClientCA is not the required field, but https://docs.openshift.com/container-platform/4.17/rest_api/config_apis/apiserver-config-openshift-io-v1.html#spec-servingcerts-namedcertificates-2